service/migration_manager.cc

```diff
@@ -958,18 +958,20 @@ future<> migration_manager::push_schema_mutation(const gms::inet_address& endpoi
     return _messaging.send_definitions_update(id, std::vector<frozen_mutation>{}, std::move(cm));
 }
 
+template<typename mutation_type>
 future<> migration_manager::announce_with_raft(std::vector<mutation> schema, group0_guard guard, std::string_view description) {
     assert(this_shard_id() == 0);
     auto schema_features = _feat.cluster_schema_features();
     auto adjusted_schema = db::schema_tables::adjust_schema_for_schema_features(std::move(schema), schema_features);
 
     auto group0_cmd = _group0_client.prepare_command(
-        schema_change{
-            .mutations{adjusted_schema.begin(), adjusted_schema.end()},
+        mutation_type {
```

Comment:
passing the command type as enum would be better than a template parameter, and more future ready if more sophisticated changes are needed down the road.

Reply:
It seems to me like it's more a @kbr-scylla question, since this change stems from the fact that raft commands are types, not enums.
I could translate these types into enums and vice versa only for this use case, but that'd perhaps result in even worse readability.

Reply:
@kbr-scylla do you have any input here?

Reply:
I don't know how what it would mean to pass command type as enum, different commands are different struct types, so I don't really understand.

It's out of the scope of this PR anyway.

Reply:
@kbr-scylla thanks
@bhalevy closing